src/NRediSearch/Client.cs

```diff
@@ -50,17 +50,96 @@ public enum IndexOptions
+            internal readonly double _score;
+            internal readonly string _payloadField;
+
+            public IndexDefinition(bool async = false, string[] prefixes = null,
+            string filter = null, string languageField = null, string language = null, 
+            string scoreFiled = null, double score = 1.0, string payloadField = null)
+            {
+                _async = async;
+                _prefixes = prefixes;
+                _filter = filter;
+                _languageField = languageField;
+                _language = language;
+                _scoreFiled = scoreFiled;
+                _score = score;
+                _payloadField = payloadField;
+            }
+
+            internal void SerializeRedisArgs(List<object> args)
+            {
+                args.Add("ON".Literal());
+                args.Add(_type);
+                if (_async)
+                {
+                    args.Add("ASYNC".Literal());
+                }
+                if (_prefixes?.Length > 0) 
+                {
+                    args.Add("PREFIX".Literal());
+                    args.Add(_prefixes.Length.ToString());
+                    args.AddRange(_prefixes);
+                }
+                if (_filter != null) 
+                {
+                    args.Add("FILTER".Literal());
+                    args.Add(_filter);
+                }                
+                if (_languageField != null) {
+                    args.Add("LANGUAGE_FIELD".Literal());
+                    args.Add(_languageField);      
+                }                
+                if (_language != null) {
+                    args.Add("LANGUAGE".Literal());
+                    args.Add(_language);      
+                }                
+                if (_scoreFiled != null) {
+                    args.Add("SCORE_FIELD".Literal());
+                    args.Add(_scoreFiled);      
+                }                
+                if (_score != 1.0) {
+                    args.Add("SCORE".Literal());
+                    args.Add(_score.ToString());      
+                }
+                if (_payloadField != null) {
+                    args.Add("PAYLOAD_FIELD".Literal());
+                    args.Add(_payloadField);      
+                }
+            }
+
+        }
+
         public sealed class ConfiguredIndexOptions
         {
             // This news up a enum which results in the 0 equivalent.
             // It's not used in the library and I'm guessing this isn't intentional.
             public static IndexOptions Default => new IndexOptions();
 
             private IndexOptions _options;
+            private IndexDefinition _definition;
             private string[] _stopwords;
-            public ConfiguredIndexOptions(IndexOptions options = IndexOptions.Default)
+            public ConfiguredIndexOptions(IndexOptions options = IndexOptions.Default, IndexDefinition definition = null)
```

Comment:
this is a binary-incompatible change; if possible, I would rather avoid hard breaks - could we do instead:

``` c#
public ConfiguredIndexOptions(IndexOptions options)
  =>ConfiguredIndexOptions(options, null);

public ConfiguredIndexOptions(IndexOptions options = IndexOptions.Default, IndexDefinition definition = null)
{...}
```

which will have the same API usage, but isn't a break

Reply:
:+1: